### PR TITLE
Add Freyr keyboard

### PR DESCRIPTION
**Pull request by vuhopkep**
## Description

Add new Freyr tkl keyboard
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/freyr/freyr.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    writePinHigh(B3);
+  }
+  if (IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK)) {
+    // Turn Scroll Lock LED on
+    writePinLow(B7);
+  } else {
+    // Turn Scroll Lock LED off
+    writePinHigh(B7);
+  }
+}
```
Comment (drashna) [inline keyboards/freyr/freyr.c]:
```suggestion
    led_set_user(usb_led);
}
```

File: keyboards/freyr/freyr.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ */
+#include "freyr.h"
+
+
+void matrix_init_kb(void) {
+  matrix_init_user();
+  led_init_ports();
+};
+
+void matrix_scan_kb(void) {
```
Comment (drashna) [inline keyboards/freyr/freyr.c]: This can be removed, since it's not doing anything different.

File: keyboards/freyr/readme.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# freyr
+
+![Freyr](https://i.imgur.com/MeOrzoI.jpg)
+
+Freyr is a tenkeyless keyboard,
```
Comment (drashna) [inline keyboards/freyr/readme.md]: If this matches the community TKL layout, you may want to consider adding community layout support for the board.

https://github.com/qmk/qmk_firmware/tree/master/layouts/default/tkl_iso
https://github.com/qmk/qmk_firmware/tree/master/layouts/default/tkl_ansi


https://docs.qmk.fm/#/feature_layouts

**Comment by vuhopkep** [comment]
A small quantily of Freyr keyboard will be on sale on reddit. So i think it deserve to be on top folder

**Comment by drashna** [comment]
Okay, not a problem, then.  It's just a standard question, since .... we have just a few keyboards in the repo. :D 
And good luck with the sale!

File: keyboards/freyr/freyr.c (hunk 1 above)
Comment (vuhopkep) [reply keyboards/freyr/freyr.c]: done

File: keyboards/freyr/freyr.c (hunk 2 above)
Comment (vuhopkep) [reply keyboards/freyr/freyr.c]: done :D

File: keyboards/freyr/readme.md (hunk 3 above)
Comment (vuhopkep) [reply keyboards/freyr/readme.md]: okie, i added 2 layouts

File: keyboards/freyr/info.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "freyr", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "width": 21.25, 
+    "height": 6.5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/freyr/info.json]: This needs to match the layout used in the keyboard's header file, actually: 

```suggestion
        "LAYOUT_all": {
```

Comment (vuhopkep) [reply keyboards/freyr/info.json]: done

File: keyboards/freyr/info.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (drashna) [inline keyboards/freyr/info.json]: Could you add the TKL ansi and is layouts here, as well?

Comment (vuhopkep) [reply keyboards/freyr/info.json]: Oh, sorry about this.
i will add right now

File: keyboards/freyr/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+enum hnah_layers{
+  _QWERTY,
+  _LOWER
+};
+
+#define LOWER MO(_LOWER)
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT_tkl_iso( \
```
Comment (fauxpark) [inline keyboards/freyr/keymaps/default/keymap.c]: The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

Comment (vuhopkep) [reply keyboards/freyr/keymaps/default/keymap.c]: Okie, thank you.
I'll deleted it

**Pull request merged**